component/async/component.go

```diff
@@ -168,38 +168,32 @@ func (c *Component) processing(ctx context.Context) error {
 		return fmt.Errorf("failed to get consumer channels: %w", err)
 	}
 
-	failCh := make(chan error)
-
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				log.Info("closing consumer")
-				failCh <- cns.Close()
-			case msg := <-chMsg:
-				log.Debug("New message from consumer arrived")
-				c.processMessage(msg, failCh)
-			case errMsg := <-chErr:
-				failCh <- fmt.Errorf("an error occurred during message consumption: %w", errMsg)
-				return
+	for {
+		select {
+		case msg := <-chMsg:
+			log.FromContext(msg.Context()).Debug("consumer received a new message")
+			err := c.processMessage(msg)
+			if err != nil {
+				return err
```

Comment:
thats an interesting case ... Do we want to stop processing at the error ? Or just ignore ? Also it would be interesting to see how the retry will work. (for now it was only used for connection failures). What scenario i am thinking of is the consumer error during deserialising of a poisonous message. 
Having said that ... should we maybe leave this open in a property ? And the default behavior could be the one you are implementing now (e.g. fail fast)

Reply:
The behavior did not change in this PR, it was doing the same before.

I did not even change the behavior on context error (we are ignoring the error there, see a couple lines above).

I would suggest to not change behavior in this PR but address it in a different PR; for example you could have a channel for processing messages and fail the whole thing only in case of non-processing errors.

Reply:
I agree not to chnage anything in this PR...
Since Kafka is a stream and not a queue the problems that you mentioned should be handled by the consumer code. No DLQ here I am afraid.
If we ignore an error then the partition iterator will move forward and the message will be "lost"...

Reply:
yeap good point... missed the lines above :) .
I see now the behavior does not change .